classes/PaymentModule.php

```diff
@@ -1274,18 +1307,211 @@ protected function createOrderCartRules(
         return $cart_rules_list;
     }
 
-    private function addShipmentToOrder(Order $order)
-    {
-        /** @var FeatureFlagStateCheckerInterface $featureFlagManager */
-        $featureFlagManager = $this->get(FeatureFlagStateCheckerInterface::class);
+    private function createOrderFromCartForMultiShipment(
+        Cart $cart,
+        Currency $currency,
+        $productList,
+        $addressId,
+        $context,
+        $reference,
+        $secure_key,
+        $payment_method,
+        $name,
+        $dont_touch_amount,
+        $amount_paid,
+        $warehouseId,
+        $cart_total_paid,
+        $debug,
+        $order_status,
+        $id_order_state,
+        $carrierId = null
```

Comment:
Maybe we could put some types here 🤔
Yeah it's legacy I know, but maybe it could be nice.
I'm not blocking your PR for this anyway!